### PR TITLE
Expand `!info` to include users not in the guild

### PR DESCRIPTION
**Pull request by Scott-Caldwell**
Utilizes the Discord REST client to retrieve user information for all users, including users not in the guild.

Additionally:
* Displays the user's online status if available

image

### CODE REVIEW

File: Modix.Services/Core/UserService.cs [hunk 1]
```diff
@@ -124,22 +140,44 @@ public async Task<GuildUserSummary> GetGuildUserSummaryAsync(ulong guildId, ulon
...
-                        data.Nickname = user.Nickname;
-                    data.LastSeen = DateTimeOffset.Now;
-                }))
+                         data.Username = user.Username;
+                     if (user.DiscriminatorValue != 0)
+                         data.Discriminator = user.Discriminator;
+                     if ((user.Username != null) && (user.DiscriminatorValue != 0))
+                         data.Nickname = user.Nickname;
+                     data.LastSeen = DateTimeOffset.Now;
+                 }))
```
Comment (jrusbatch) [inline Modix.Services/Core/UserService.cs]: nit: looks like some extra whitespace was added?

Comment (Scott-Caldwell) [reply Modix.Services/Core/UserService.cs]: Fixed

**Comment by jmazouri** [comment]
@Scott-Caldwell Before we merge, I pushed some changes I felt made sense - what are your thoughts?

**Comment by Scott-Caldwell** [comment]
Yup, I think that makes sense - `EphemeralUser` and `UserInformation` were doing essentially the same thing. I should have thought of it myself.

**Pull request merged**